src/cpp/types/topic/Topic.cpp

```diff
@@ -48,11 +51,17 @@ const std::string& Topic::topic_type() const
     return topic_type_;
 }
 
+bool Topic::topic_with_key() const
+{
+    return topic_with_key_;
+}
+
 bool Topic::operator ==(
         const Topic& other) const
 {
 
-    return topic_name_ == other.topic_name_ && topic_type_ == other.topic_type_;
+    return topic_name_ == other.topic_name_ && topic_type_ == other.topic_type_ &&
```

Comment:
NIT: I think this is okey, but we have to take care, Topic with and without key is an incoherent error, the key does not uniquely represent a topic.

Reply:
:+1: